src/packages/database/database_manager.py

```diff
@@ -53,6 +53,59 @@ class property, and the DatabaseManage.query() method used to perform a query.
 
     """
 
+    @classmethod
+    @CONFIG_MARKER
+    def rehash_handler(cls, data: Dict):
+        """
+        Apply new configuration data
+
+        Args:
+            data (typing.Dict): new configuration data to apply.
+
+        """
+        cls._config = data
+
+    _config: ClassVar[Dict] = {}
+
+    @classmethod
+    @CONFIG_MARKER
+    def validate_config(cls, data: Dict):
+        """
+        Validate database portion of the configuration file
+
+        Args:
+            data(typing.Dict): configuration object
+        """
+        try:
+            module_config = data['database']
+        except (KeyError, AttributeError) as ex:
+            raise ValueError("No Database configuration in file.") from ex
+
+        # Require all values to be set
+        for setting in module_config.items():
+            if setting is None:
+                raise ValueError(f"{setting} is required for instantiation")
+
+        # Host
+        if not isinstance(module_config['host'], str):
+            raise ValueError("host must be a string.")
```

Comment:
[Question] Perhaps these error messages should be a tad more specific? Unlike logging, we don't have any context here.
We could possibly have several config options by the name of `host` under different categories.

Reply:
the erroring module will be in the stacktrace, though the most direct thing we can do here is have `__name__` embedded into the error strings or something.

Thoughts?